src/vl-map-action.js

```diff
@@ -80,30 +117,38 @@ export class VlMapAction extends vlElement(HTMLElement) {
    */
   actionChanged() {
     const event = new Event(VlMapAction.NEW_ACTION_EVENT_NAME);
-    this.parentElement.dispatchEvent(event);
+    this._mapElement.dispatchEvent(event);
   }
 
   _layerChangedCallback() {
-    this._computeAction(this._map, this.layer);
+    this._mapElement.ready.then(() => {
+      if (this.layer) {
+        const action = this._createAction(this.layer);
+        if (action) {
+          this._mapElement.addAction(action);
+          this.actionChanged();
+          if (this._active) {
+            action.activate();
+          }
+          this._action = action;
+        }
+      }
```

Comment:
Ik zou dit extracten zodat je hier een functie met een beschrijvende naam kunt oproepen (denk dat ik dit ook op de andere PR heb aangehaald).

Reply:
Ik denk dat dit niet meer relevant is voor deze PR, dus markeer het als resolved. Anders geef je maar een seintje.

Reply:
ja met da ge PR's precies op een of andere manier op mekaar laat voortbouwen is da vervelend voor de reviewers soms. 
dus moest er een mogelijkheid zijn om da anders te doen ...

Reply:
Wel @SanderKleykens was hier eigenlijk te vroeg met reviewen, deze wijzigingen zijn momenteel geen onderdeel meer van de PR omdat ik de PR's pas doorschuif als de vorige PR goedgekeurd werd en bijgevolg de merge uitgevoerd werd waardoor alleen de echte wijzigingen voor jullie zichtbaar zijn.